src/collections/maple_tree/mod.rs

```diff
@@ -200,18 +213,90 @@ where
                 M::Some(v) => {
                     return Some(v);
                 }
-                M::None => {}
+                M::None => panic!(
+                    "SparseLeaf - remove() None value found associated to a valid key, aborting."
+                ),
             }
         }
+
         None
     }
 
+    // This function is used to help verify the validity of the entire tree
+    // this function returns true if all keys within the SparseLeaf are within the bounds
+    // min to max or equal to min or max or the SparseLeaf node is empty
+    // otherwise this returns false
+    pub fn check_min_max(&mut self, min: &K, max: &K) -> bool {
```

Comment:
Should this be fn check_min_max (no pub) and then the pub is "pub verify" ..? I guess we'll come to that when we get to it though because verify may be different with a tree involved.

Better idead, could be this become

pub fn get_max()
pub fn get_min()
pub fn check_bounds() -> {
    self.get_max ...
    self.get_min ...
}

Seems to have better re-usability and will work regardless of how we design a fn verify later?

Reply:
yeah, your idea sounds good and I will implement this today and send it through this afternoon.